anesthetic/weighted_pandas.py

```diff
@@ -157,24 +143,16 @@ def compress(self, nsamples=None):
             compression). If <=0, then compress so that all weights are unity.
 
         """
-        i = compress_weights(self.weight, self._rand, nsamples)
+        i = compress_weights(self.weights, self._rand, nsamples)
         data = np.repeat(self.values, i, axis=0)
-        index = np.repeat(self.index.values, i)
+        index = self.index.repeat(i)
```

Comment:
Using `index.repeat` instead of `np.repeat` keeps the same `MultiIndex` structure. With `np.repeat` the `'#'` and `weights` index were turned into a tuple such that `compress` returned:
```python
> print(ns_compressed[:5])

                                   x0        x1      logL  logL_birth  nlive
(476, 0.00975239917398711)  -0.236910 -0.284601 -4.606267   -6.359672  125.0
(495, 0.016423622996336068) -0.004446  0.162164 -3.933665 -556.623795  125.0
(521, 0.03131786144184918)  -0.035268 -0.094386 -3.081024   -8.523751  125.0
(554, 0.054268694486099424) -0.115377 -0.128187 -2.268316   -2.823167  125.0
(568, 0.07583551199263458)   0.263529  0.117131 -1.822142   -1.935448  125.0
```

now it returns:
```python
> print(ns_compressed[:5])

           x0        x1      logL  logL_birth  nlive
#                                                   
476 -0.236910 -0.284601 -4.606267   -6.359672  125.0
495 -0.004446  0.162164 -3.933665 -556.623795  125.0
521 -0.035268 -0.094386 -3.081024   -8.523751  125.0
554 -0.115377 -0.128187 -2.268316   -2.823167  125.0
568  0.263529  0.117131 -1.822142   -1.935448  125.0
```

where  
```python
ns = NestedSamples(root="anesthetic/tests/example_data/pc")
ns = ns[['x0', 'x1', 'logL', 'logL_birth', 'nlive']]
ns_compressed = ns.compress(1000)
```

Reply:
Nice!